src/librustdoc/json/types.rs

```diff
@@ -55,11 +55,16 @@ pub struct ItemSummary {
 
 #[derive(Clone, Debug, Serialize, Deserialize)]
 pub struct Item {
+    /// The unique identifier of this item. Can be used to find this item in various mappings.
+    pub id: Id,
     /// This can be used as a key to the `external_crates` map of [`Crate`] to see which crate
     /// this item came from.
     pub crate_id: u32,
     /// Some items such as impls don't have names.
     pub name: Option<String>,
+    /// Whether this item is meant to be omitted from the generated documentation due to `#doc(hidden)`,
+    /// because it is private, or because it was inlined.
+    pub stripped: bool,
```

Comment:
I still prefer to omit stripped items altogether. I won't block on it because I really want to see this in nightly 😆 but it should be removed before stabilization; rustdoc should not have a stable way to introspect private items from other crates.

Reply:
`--document-private-items` is really useful for generating internal docs, and I think this feature should support it, even though it technically provides a way to do what you say.

Reply:
If you pass `--document-private-items`, the items won't be stripped (and in fact you'll lose the ability to distinguish between private and public items). This is more like 'unconditionally document private items'.